CMSgov/onemac
# Oy2 10464

Story: https://qmacbis.atlassian.net/browse/OY2-10464
Endpoint: https://d2751bya3a0tdv.cloudfront.net

### Details

Allow users to view a read-only version of other users' profiles. Links to this view are accessible in the User Management table and the Dashboard table.

### Changes

- Add read-only capability to the `UserPage` and `PhoneNumber` components
- Add a routing entry for **all users** to access this view at `/profile/[user id]` where (for now at least) that ID is the user's email address
- Remove the Email Address column from the User Management table
- Change the Name column there into a link to that user's profile view
- Change the State Submitter [sic] column into a link to that user's profile view

### Implementation Notes

- Much of the local seed data for submissions contains garbage in the `user` field - either email addresses not present in our user profile seed data, or different names with the same email. Should that be cleaned up? Seems like a big and not-so-valuable undertaking.

### Test Plan

1. Login as a state submitter. View your user profile through the normal means (navbar menu) and ensure it still works as desired.
2. Return to the dashboard and click **your** name in the State Submitter column. It should take you to **your** profile, and it should work exactly the same as if you accessed it the old way.
3. Repeat step 1 as a state admin.
4. Return to the User Management table and verify that each user in the table is linked to their own profile, which is in a read-only mode. You should not be able to revoke (no X button) or request (no plus button) accesses on behalf of that user, nor edit their phone number. If no phone number has been set for a user, it should read "N/A".
5. Repeat steps 3 and 4 as a CMS Role Approver and as a System Admin user.
